lib/features/calendar/providers/calendar_provider.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+class CalendarProvider with ChangeNotifier {
```

Comment:
Leave only these variables:
todayDate (get)
firstDay (get)
formattedTodayDate (get)
formattedSelectedDate (get)
selectedMonth
selectedYear
selectedDate
years
months

And only these functions:
changeMonthDate(int index) - sets currentMonth by index from months
changeYearDate(int index) - sets currentYear by index from years
pickDate() - sets selectedDate from currentMonth and currentYear